tools/models/model_factory.py

```diff
@@ -0,0 +1,168 @@
+        print(
+            (
+                "[WW] WORLD_SIZE not defined in env, "
+                + "falling back to SLURM_NTASKS."
+            ),
+            file=sys.stderr,
+        )
+        world_size = int(os.environ["SLURM_NTASKS"])
+    else:
+        raise RuntimeError("cannot determine WORLD_SIZE")
+    return world_size
+
+def get_VAE_encoder_kwargs(io_config, model_config):
+    """Create encoder kwargs dictionary from configs"""
+    return {
+        "ae_config": "non_deterministic",
+        "z_dim": model_config.latent_space_dims,
+        "input_dim": io_config.ps_dims,
+        "conv_layer_config": [16, 32, 64, 128, 256, 608],
+        "conv_add_bn": False,
+        "fc_layer_config": [544],
+    }
+
+def get_VAE_decoder_kwargs(io_config, model_config):
+    """Create decoder kwargs dictionary from configs"""
+    return {
+        "z_dim": model_config.latent_space_dims,
+        "input_dim": io_config.ps_dims,
+        "initial_conv3d_size": [16, 4, 4, 4],
+        "add_batch_normalisation": False,
+        "fc_layer_config": [1024],
+    }
+
+def load_objects(rank, io_config, model_config, world_size):
+    """Load and initialize model objects, optimizer, and scheduler."""
+    
+    # Get configuration
+    config = model_config.config
+    
+    # Get model parameters
+    VAE_encoder_kwargs = get_VAE_encoder_kwargs(io_config, model_config)
+    VAE_decoder_kwargs = get_VAE_decoder_kwargs(io_config, model_config)
+    
+    # Initialize loss function
+    loss_fn_for_VAE = MAPPING_TO_LOSS[
+        model_config.config["loss_function"]
+    ](**model_config.config["loss_kwargs"])
+
+    # Initialize VAE
+    VAE_obj = VAE(
+            encoder=Encoder,
+            encoder_kwargs=VAE_encoder_kwargs,
+            decoder=Conv3DDecoder,
+            z_dim=model_config.latent_space_dims,
+            decoder_kwargs=VAE_decoder_kwargs,
+            loss_function=loss_fn_for_VAE,
+            property_="momentum_force",
+            particles_to_sample=io_config.number_of_particles,
+            ae_config="non_deterministic",
+            use_encoding_in_decoder=False,
+            weight_kl=model_config.config["lambd_kl"],
+            device=rank,
+    )
+
+    # Initialize inner model
+    inner_model = INNModel(
+            ndim_tot=config["ndim_tot"],
+            ndim_x=config["ndim_x"],
+            ndim_y=config["ndim_y"],
+            ndim_z=config["ndim_z"],
+            loss_fit=fit,
+            loss_latent=MMD_multiscale,
+            loss_backward=MMD_multiscale,
+            lambd_predict=config["lambd_predict"],
+            lambd_latent=config["lambd_latent"],
+            lambd_rev=config["lambd_rev"],
+            zeros_noise_scale=config["zeros_noise_scale"],
+            y_noise_scale=config["y_noise_scale"],
+            hidden_size=config["hidden_size"],
+            activation=config["activation"],
+            num_coupling_layers=config["num_coupling_layers"],
```

Comment:
These params should be from a dictionary in model_config. Most of them already are, maybe this could be renamed (more descriptive than "config", maybe "inner_model_config"). The details can then also be left out here, by just passing `**model_config.inner_model_config` to the inner model ctor.